src/sqlfluff/core/linter/linted_path.py

```diff
@@ -0,0 +1,118 @@
+"""Defines the LintedPath class.
+
+This stors the idea of a collection of linted files at a single start path
+"""
+
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Union,
+    overload,
+)
+from typing_extensions import Literal
+
+
+from sqlfluff.core.errors import (
+    CheckTuple,
+)
+from sqlfluff.core.parser.segments.base import BaseSegment
+
+from sqlfluff.core.linter.linted_file import LintedFile
+
+
+class LintedPath:
+    """A class to store the idea of a collection of linted files at a single start path."""

```

Comment:
Nitpick: Would "directory" be a better term than "path" here?

https://unix.stackexchange.com/questions/131561/what-is-the-difference-between-path-and-directory/131585

Reply:
yes - good point.